cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -101,13 +103,24 @@ func Run(gadgetRoot, device string, options Options) error {
 		return fmt.Errorf("cannot create the partitions: %v", err)
 	}
 
-	// generate key externally so multiple encrypted partitions can use the same key
+	// generate keys externally so multiple encrypted partitions can use the same key
 	var key partition.EncryptionKey
+	var rkey partition.RecoveryKey
+
+	useRecoveryKey := options.RecoveryKeyFile != ""
```

Comment:
Do we have a use case for *not* using a recovery-key-file ? It seems like something we want to dis-encourage (at least for now) so I would do something like: https://paste.ubuntu.com/p/tdjrN6pkQK/ and error if no recovery key file is set.

Reply:
I talked to Chris and he suggested to enforce the recovery-key so I will go with the above patch.